### ISSUE TITLE
Slow

### ISSUE THREAD

**Issue by colt05-usr** [valarnin/NTRRemoteDebugger#20]
Hello!
When searching memory, this goes slow. Any way to speed it up?
Thanks.

**Comment by phecdaDia** [comment]
What are the criteria you're looking for? DataType, Value, Memregion(s), ...

**Comment by colt05-usr** [comment]
All of the Memregions, that's probably why.

**Comment by colt05-usr** [comment]
Also, do I need to click Open again to do the next search?

**Comment by phecdaDia** [comment]
No, you should be able to just change the value you're searching for and refine the results.

**Comment by valarnin** [comment]
There's not really a way to speed up memory searches beyond what's already in place, unfortunately. The 3DS's wifi is slow, and we can't ask the 3DS to search for a specific byte set in memory because NTR CFW does not have such a feature. Unfortunately, these features are in the .bin for NTR and not in the actual C++ source, so we can't contribute a feature back to NTR for this either.

**Comment by JustForNothing** [comment]
But to make it faster, you can't dump the 1st memory dump, and then search the value directly in the dumped file ?
Should be faster.

**Comment by valarnin** [comment]
To clarify, the way it works is that there's a threshold where if there are more than [200 results](https://github.com/valarnin/NTRRemoteDebugger/blob/master/NTRDebuggerTool/Remote/NTRRemoteConnection.cs#L226) currently found for a search, it will dump the entire memory region selected to search, then compare only the valid addresses from that region. Otherwise, it dumps memory for each address in the results collection and compares them.

We can't keep comparing values from the first memory dump, that wouldn't have any updated values in it.

This parameter should really be moved to the configuration settings GUI, but I won't have time to work on this project in general until the new year unfortunately.
